hcl/json/structure.go

```diff
@@ -324,6 +327,8 @@ func (b *body) collectDeepAttrs(v node, labelName *string) ([]*objectAttr, hcl.D
 	var attrs []*objectAttr
 
 	switch tv := v.(type) {
+	case *nullVal:
```

Comment:
I'm not sure how to test this case, because I'm not 💯 sure how to repro this code path, happy to take suggestions. (The first commit solves the specific case that I was looking for)

Reply:
This case arises if the block header schema has one or more labels defined. In HCL native syntax a block with one label looks like:

```
block "foo" {
}
```

In JSON that becomes a multi-level structure including one extra object level to represent the label:

```json
{
  "block": {
    "foo": {}
  }
}
```

Or, the extra level can be wrapped in an array to force order to be retained:

```json
{
  "block": [
    {
      "foo": {}
    }
  ]
}
```

This `unpackBlock` function, then, is where we would deal with either that single labels object being null or one of the objects in the labels array being null. The other code path you updated in this PR then deals with the final level of object that represents the body of the block itself.

Reply:
Awesome - updated this to add a test for the nested blocks - As of now it detects the block with no nested labels.